divi/src/miner.cpp

```diff
@@ -121,35 +150,25 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn, CWallet* pwallet,
     // Create coinbase tx
     CMutableTransaction txNew = CreateCoinbaseTransaction(scriptPubKeyIn);
     
-    AddTransactionToBlock(pblock, pblocktemplate, txNew);
+    AddTransactionToBlockAndSetDefaultFees(*pblock, pblocktemplate, txNew);
 
     // ppcoin: if coinstake available add coinstake tx
     static int64_t nLastCoinStakeSearchTime = GetAdjustedTime(); // only initialized at startup
 
     if (fProofOfStake) {
         boost::this_thread::interruption_point();
 
-        pblock->nTime = GetAdjustedTime();
-        CBlockIndex* pindexPrev = chainActive.Tip();
-        pblock->nBits = GetNextWorkRequired(pindexPrev, pblock);
+        
         CMutableTransaction txCoinStake;
 
+        SetRequiredWork(*pblock);
```

Comment:
Moving the code + Extracting the function + Replacing the function should be three separate commits. So that it's easier to review when it happens